In [10]:
import numpy as np
import pandas as pd
import os
from utils import make_model_sequential,load_images_with_labels,split_train,final_model
from sklearn.metrics import accuracy_score


TRAIN_IMAGE_PATH=os.path.join(os.curdir,"processed data","train_images.npy")
TRAIN_LABELS_PATH=os.path.join(os.curdir,"processed data","train_labels.npy")
TEST_IMAGE_PATH=os.path.join(os.curdir,"processed data","test_images.npy")

X_train_full,Y_full,X_test=load_images_with_labels(TRAIN_IMAGE_PATH,TRAIN_LABELS_PATH,TEST_IMAGE_PATH)
X_train,X_valid,y_train,y_valid=split_train(X_train_full,Y_full,validation_size=0.2,random_state=42)


In [2]:
np.unique(y_train)

array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16,
       17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33,
       34, 35, 36, 37, 38, 39, 40, 41, 42])

In [6]:
model=make_model_sequential()
model.summary()

Model: "sequential_1"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
conv2d_6 (Conv2D)            (None, 26, 26, 32)        2432      
_________________________________________________________________
conv2d_7 (Conv2D)            (None, 22, 22, 32)        25632     
_________________________________________________________________
max_pooling2d_3 (MaxPooling2 (None, 11, 11, 32)        0         
_________________________________________________________________
dropout_1 (Dropout)          (None, 11, 11, 32)        0         
_________________________________________________________________
conv2d_8 (Conv2D)            (None, 9, 9, 64)          18496     
_________________________________________________________________
conv2d_9 (Conv2D)            (None, 7, 7, 64)          36928     
_________________________________________________________________
max_pooling2d_4 (MaxPooling2 (None, 3, 3, 64)         

In [7]:
model.load_weights("Initial_Model_weights.h5")

In [8]:
#Loading the test labels

test=pd.read_csv(os.path.join(os.curdir,"data","Test.csv"))
test.head()
y_test=test["ClassId"].values

In [9]:
y_preds=model.predict(X_test)
y_preds=np.argmax(y_preds, axis=-1)
accuracy=accuracy_score(y_test, y_preds)
print("Accuracy on the test set-------> ",accuracy)

Accuracy on the test set------->  0.9476642913697546


<h2>We achieved 94.7% accuracy on the test set using the first model</h2>

In [11]:
model=final_model()
model.summary()

Model: "sequential_2"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
conv2d_10 (Conv2D)           (None, 30, 30, 32)        896       
_________________________________________________________________
conv2d_11 (Conv2D)           (None, 30, 30, 32)        9248      
_________________________________________________________________
max_pooling2d_5 (MaxPooling2 (None, 15, 15, 32)        0         
_________________________________________________________________
conv2d_12 (Conv2D)           (None, 15, 15, 64)        18496     
_________________________________________________________________
conv2d_13 (Conv2D)           (None, 15, 15, 64)        36928     
_________________________________________________________________
max_pooling2d_6 (MaxPooling2 (None, 7, 7, 64)          0         
_________________________________________________________________
conv2d_14 (Conv2D)           (None, 7, 7, 128)        

In [12]:
model.load_weights("Final_Model_weights.h5")

In [13]:
y_preds=model.predict(X_test)
y_preds=np.argmax(y_preds, axis=-1)
accuracy=accuracy_score(y_test, y_preds)
print("Accuracy on the test set-------> ",accuracy)

Accuracy on the test set------->  0.9595407759303246


<h2>We achieved 95.95% accuracy on the test set using the final model</h2>